pkg/services/object/get/ec.go

```diff
@@ -192,7 +184,77 @@ func (s *Service) restoreFromECParts(ctx context.Context, cnr cid.ID, parent oid
 		)
 	}
 
-	return object.Object{}, apistatus.ErrObjectNotFound
+	return apistatus.ErrObjectNotFound
+}
+
+func (s *Service) copySplitECObject(ctx context.Context, dst ObjectWriter, cnr cid.ID, parent oid.ID, sTok *session.Object,
+	rules []iec.Rule, sortedNodeLists [][]netmap.NodeInfo, fromRule int, linker object.Object) error {
+	parentHdr := linker.Parent()
+	if parentHdr == nil {
+		return fmt.Errorf("%w: missing parent header", errInvalidSizeSplitLinker)
+	}
+
+	var l object.Link
+	if err := linker.ReadLink(&l); err != nil {
+		return fmt.Errorf("%w: %w", errInvalidSizeSplitLinker, err)
+	}
+	sizeSplitParts := l.Objects()
+	if len(sizeSplitParts) == 0 {
+		return fmt.Errorf("%w: empty part list", errInvalidSizeSplitLinker)
+	}
+
+	if err := dst.WriteHeader(parentHdr); err != nil {
+		return fmt.Errorf("%w: write parent header: %w", errStreamFailure, err)
+	}
+
+	var partTimeout time.Duration
+	if deadline, ok := ctx.Deadline(); ok {
+		// don't take into account that last part may be tiny
+		if partTimeout = time.Until(deadline) / time.Duration(len(sizeSplitParts)); partTimeout <= 0 {
```

Comment:
When do you expect this condition to be met? Why do we care at all? What if one object is fetching slowly, but others are OK?

Reply:
> When do you expect this condition to be met?

if context is almost done. Too small timeout is set by the client, or network took most of it

> Why do we care at all? What if one object is fetching slowly, but others are OK?

The delay for each part is most likely the same, so it's unlikely OK to download everything before the deadline. Although, in rare cases, it might be possible. In general, we don't have to do this